wormhole-connect/src/components/DemoAppHeader/index.tsx

```diff
@@ -0,0 +1,82 @@
+import './styles.css';
+import React, { useEffect } from 'react';
+import { useState } from 'react';
+import { WormholeConnectConfig } from 'config/types';
+
+interface Props {
+  onCustomConfigChange: (config: WormholeConnectConfig | undefined) => void;
+}
+
+const loadInitialConfig = (): string => {
+  const params = new URLSearchParams(window.location.search);
+  const configQuery = params.get('config');
+  const configCached = localStorage.getItem(LOCAL_STORAGE_KEY);
+  return configQuery || configCached || '';
+};
+
+const setUrlQueryParam = (configInput: string) => {
+  const url = new URL(window.location.toString());
+  url.searchParams.set('config', configInput);
+  history.replaceState({}, '', url.toString());
+};
+
+const LOCAL_STORAGE_KEY = 'wormhole-connect:demo:custom-config';
+
+export default function DemoAppHeader(props: Props) {
+  const [customConfigOpen, setCustomConfigOpen] = useState(false);
+  const [customConfigInput, setCustomConfigInput] = useState(
+    loadInitialConfig(),
+  );
+
+  const updateCustomConfig = (e: any) => {
+    const input = e.target.value;
+    setCustomConfigInput(input);
+  };
+
+  const emitCustomConfig = () => {
+    localStorage.setItem(LOCAL_STORAGE_KEY, customConfigInput);
+    setUrlQueryParam(customConfigInput);
+
+    try {
+      const parsed = eval(`(function() {
```

Comment:
I'm parsing the config as Javascript code, not merely JSON, so we can put in callback functions and stuff. I figure it's safe to do this for local dev server & internal Netlify builds. but it would technically be safer to only accept JSON and use `JSON.parse` here instead.

Reply:
nvm I got scared and changed this to vanilla JSON.parse